nemo_aligner/algorithms/supervised.py

```diff
@@ -133,6 +143,19 @@ def train_single_step(self, batch):
 
         return loss_mean, trainer_metrics | metrics
 
+    @torch.no_grad()
+    def run_generation(self, batch):
+        context_tokens, context_lengths = batch["contexts"], batch["context_lengths"]
+        max_prompt_length = context_lengths.max().item()
+        max_response_length = self.model.get_inference_params()["length_params"]["max_length"]
+        max_length = max_prompt_length + max_response_length
+        # nemo requires us to pad the response length up before we do anything
+        context_tokens = torch.nn.functional.pad(context_tokens, (0, max_length), value=self.model.tokenizer.eos_id)
```

Comment:
is it possible to use the function here? https://github.com/NVIDIA/NeMo-Aligner/blame/09b3dccaf4aebfb515e5d531ee2cae4416c67ff3/nemo_aligner/utils/utils.py#L292